src/plugins/vis_augmenter/public/saved_augment_vis/_saved_augment_vis.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @name SavedAugmentVis
+ *
+ * @extends SavedObject.
+ */
+import { get } from 'lodash';
+import {
+  createSavedObjectClass,
+  SavedObject,
+  SavedObjectOpenSearchDashboardsServices,
+} from '../../../saved_objects/public';
+import { IIndexPattern } from '../../../data/public';
+import { extractReferences, injectReferences } from './saved_augment_vis_references';
+
+export function createSavedAugmentVisClass(services: SavedObjectOpenSearchDashboardsServices) {
+  const SavedObjectClass = createSavedObjectClass(services);
+
+  class SavedAugmentVis extends SavedObjectClass {
+    public static type: string = 'augment-vis';
+    public static mapping: Record<string, string> = {
+      description: 'text',
+      pluginResourceId: 'text',
+      visId: 'keyword',
+      visLayerExpressionFn: 'object',
```

Comment:
Is there a reason you want an `object` for this? Other saved objects I've seen just use `text` and store a JSON serialized blob. But having a well defined object may actually be better, depending on how you plan to use it.

Reply:
Good point. I think I will switch to `text` and store a JSON serialized blob. My concern with having a well-defined object for this is complications around fluidity on the `args` field, and I don't think there's any reason to make it more typesafe here.